honeybee/aperture.py

```diff
@@ -0,0 +1,506 @@
+            offset: A number for the distance to louvers from this aperture.
+                Default is 0 for no offset.
+            angle: A number for the for an angle to rotate the louvers in degrees.
+                Default is 0 for no rotation.
+            contour_vector: A Vector3D for the direction along which contours
+                are generated. Default is Z-Axis, which generates horizontal louvers.
+            flip_start_side: Boolean to note whether the side the louvers start from
+                should be flipped. Default is False to have contours on top or right.
+                Setting to True will start contours on the bottom or left.
+            indoor: Boolean for whether louvers should be generated facing the
+                opposite direction of the aperture normal (typically meaning
+                indoor geometry). Default: False.
+
+        Returns:
+            louvers: A list of Shade objects that cover the Aperture.
+        """
+        angle = math.radians(angle)
+        louvers = []
+        ap_geo = self.geometry if indoor is False else self.geometry.flip()
+        shade_faces = ap_geo.countour_fins_by_distance_between(
+            distance, depth, offset, angle, contour_vector, flip_start_side)
+        for i, shade_geo in enumerate(shade_faces):
+            louvers.append(Shade('{}_Shd{}'.format(self.name_original, i), shade_geo))
+        return louvers
+
+    def move(self, moving_vec):
+        """Move this Aperture along a vector.
+
+        Args:
+            moving_vec: A ladybug_geometry Vector3D with the direction and distance
+                to move the face.
+        """
+        self._geometry = self.geometry.move(moving_vec)
+
+    def rotate(self, axis, angle, origin):
+        """Rotate this Aperture by a certain angle around an axis and origin.
+
+        Args:
+            axis: A ladybug_geometry Vector3D axis representing the axis of rotation.
+            angle: An angle for rotation in degrees.
+            origin: A ladybug_geometry Point3D for the origin around which the
+                object will be rotated.
+        """
+        self._geometry = self.geometry.rotate(axis, math.radians(angle), origin)
+
+    def rotate_xy(self, angle, origin):
+        """Rotate this Aperture counterclockwise in the world XY plane by a certain angle.
+
+        Args:
+            angle: An angle in degrees.
+            origin: A ladybug_geometry Point3D for the origin around which the
+                object will be rotated.
+        """
+        self._geometry = self.geometry.rotate_xy(math.radians(angle), origin)
+
+    def reflect(self, plane):
+        """Reflect this Aperture across a plane with the input normal vector and origin.
+
+        Args:
+            plane: A ladybug_geometry Plane across which the object will
+                be reflected.
+        """
+        self._geometry = self.geometry.reflect(plane.n, plane.o)
```

Comment:
Checking for instance to be plane will be helpful here because otherwise you will get an AttributeError which is not really helpful to understand the problem.

Reply:
Fair point.  I will add this check in